### PR TITLE
Consider all of the time entries visible to the user

### PR DESCRIPTION
**Pull request by Azure7111**
This should consider all of the permissions a user has, even those added by plugins.

https://community.openproject.org/work_packages/19441

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://azure7111-fix-time-entry-visibility.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://azure7111-fix-time-entry-visibility.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by floriank** [comment]
:+1:

if @NobodysNightmare has no more commentary, one may merge this.

File: app/models/time_entry.rb [hunk 1]
```diff
@@ -112,19 +112,15 @@ def self.visible_by(usr)
   end
 
   def self.earliest_date_for_project(project = nil)
-    finder_conditions = ARCondition.new(Project.allowed_to_condition(User.current, :view_time_entries))
-    if project
-      finder_conditions << ['project_id IN (?)', project.hierarchy.map(&:id)]
-    end
-    TimeEntry.minimum(:spent_on, include: :project, conditions: finder_conditions.conditions)
+    scope = TimeEntry.visible(User.current)
+    scope = scope.where(['project_id IN (?)', project.hierarchy.map(&:id)]) if project
```
Comment (myabc) [inline app/models/time_entry.rb]: @Azure7111 

``` ruby
scope.where(project_id: project.hierarchy.map(&:id))
```

should also work

Comment (Azure7111) [inline app/models/time_entry.rb]: That's exactly what I have been looking for!

**Comment by NobodysNightmare** [comment]
I re-ran two travis jobs, if they turn green it is good to merge :+1:

**Pull request merged**